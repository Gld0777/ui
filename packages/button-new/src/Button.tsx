import { StyledButton } from './ButtonStyles'
import { ButtonProps } from './types'

export default function Button({ children, ...rest }: ButtonProps) {
  return (
    <StyledButton type='button' {...rest}>
      {children}
    </StyledButton>
  )
}

Button.defaultProps = {
  size: 'md',
  variant: 'filled',
  color: 'primary',
  fullwidth: false
}
