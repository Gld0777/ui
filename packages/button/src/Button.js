import styled, { css } from 'styled-components'
import theme from '@lidofinance/theme'
import inject from '@lidofinance/injector'
import PropTypes from 'prop-types'

import shouldForwardProp from '@styled-system/should-forward-prop'

const APPEARANCES = {
  PRIMARY: 'primary',
}

const StyledButton = styled('button').withConfig({
  shouldForwardProp,
})`
  border: none;
  outline: none;
  white-space: nowrap;
  overflow: hidden;
  line-height: inherit;
  background-color: inherit;
  font-size: inherit;
  color: inherit;
  padding: inherit;

  background-color: ${theme.colors.main};
  font-size: 18px;
  font-weight: 500;
  font-family: inherit;
  line-height: 24px;
  color: #ffffff;
  padding: 16px 20px;
  border-radius: 6px;
  display: flex;
  align-items: center;
  justify-content: center;

  :hover {
    background-color: #009bf2;
  }

  :hover {
    cursor: pointer;
  }

  ${(props) =>
    props.isFullWidth &&
    css`
      width: 100%;
    `}

  ${(props) =>
    props.isDisabled &&
    css`
      background-color: #96cbfa;
      :hover {
        background-color: #96cbfa;
      }
    `}

    ${inject}
`

const Button = ({ children, isLoading, loadingText, ...rest }) => (
  <StyledButton {...rest}>
    {loadingText && isLoading ? loadingText : children}
  </StyledButton>
)

Button.propTypes = {
  isLoading: PropTypes.bool,
  isDisabled: PropTypes.bool,
  loadingText: PropTypes.string,
  isFullWidth: PropTypes.bool,
  leftIcon: PropTypes.func,
  rightIcon: PropTypes.func,
  variant: PropTypes.oneOf(Object.values(APPEARANCES)),
}

Button.defaultProps = {
  isLoading: false,
  isDisabled: false,
  loadingText: null,
  isFullWidth: false,
  leftIcon: null,
  rightIcon: null,
  variant: APPEARANCES.PRIMARY,
}

export default Button
