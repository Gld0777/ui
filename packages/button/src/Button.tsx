import styled, { css } from 'styled-components'

interface ButtonProps {
  isLoading?: boolean
  isDisabled?: boolean
  loadingText?: string
  isFullWidth?: boolean
  leftIcon?: React.ReactElement
  rightIcon?: React.ReactElement

  children: React.ReactElement | string
}

const StyledButton = styled.button<ButtonProps>`
  border: none;
  outline: none;
  white-space: nowrap;
  overflow: hidden;
  line-height: inherit;
  background-color: inherit;
  font-size: inherit;
  color: inherit;
  padding: inherit;

  background-color: ${({ theme }) => theme.colors.main};
  font-size: 18px;
  font-weight: 500;
  font-family: inherit;
  line-height: 24px;
  color: #ffffff;
  padding: 16px 20px;
  border-radius: 6px;
  display: flex;
  align-items: center;
  justify-content: center;

  :hover {
    background-color: #009bf2;
  }

  :hover {
    cursor: pointer;
  }

  ${(props) =>
    props.isFullWidth &&
    css`
      width: 100%;
    `}

  ${(props) =>
    props.isDisabled &&
    css`
      background-color: #96cbfa;
      :hover {
        background-color: #96cbfa;
      }
    `}
`

const Button = ({ children, isLoading, loadingText, ...rest }: ButtonProps) => (
  <StyledButton {...rest}>
    {loadingText && isLoading ? loadingText : children}
  </StyledButton>
)

export default Button
