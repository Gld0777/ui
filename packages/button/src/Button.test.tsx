import { render } from '@testing-library/react'
import 'jest-styled-components'

import Button from './Button'

it('renders correctly', () => {
  const { container } = render(<Button>Test</Button>)

  expect(container.firstChild).toMatchSnapshot()
})
