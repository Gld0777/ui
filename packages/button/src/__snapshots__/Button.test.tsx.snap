// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  border: none;
  outline: none;
  white-space: nowrap;
  overflow: hidden;
  line-height: inherit;
  background-color: inherit;
  font-size: inherit;
  color: inherit;
  padding: inherit;
  background-color: #00A3FF;
  font-size: 18px;
  font-weight: 500;
  font-family: inherit;
  line-height: 24px;
  color: #ffffff;
  padding: 16px 20px;
  border-radius: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0:hover {
  background-color: #009bf2;
}

.c0:hover {
  cursor: pointer;
}

<button
  class="c0"
>
  Test
</button>
`;
